procrastinate/psycopg2_connector.py

```diff
@@ -124,26 +155,42 @@ def _wrap_json(self, arguments: Dict[str, Any]):
 
     @contextlib.contextmanager
     def _connection(self) -> psycopg2.extensions.connection:
+        # in case of an admin shutdown (Postgres error code 57P01) we do not
+        # rollback the connection or put the connection back to the pool as
+        # this will cause a psycopg2.InterfaceError exception
         try:
-            with self._pool.getconn() as connection:
+            connection = self._pool.getconn()
+            try:
                 yield connection
-        finally:
+            except psycopg2.errors.AdminShutdown:
+                raise
+            except Exception:
+                connection.rollback()
+                raise
+            else:
+                connection.commit()
+        except psycopg2.errors.AdminShutdown:
+            raise
```

Comment:
I think those 2 lines do nothing. If we removed them, an AdminShutdown would also bubble up.